app/controllers/declaration/FiscalInformationController.scala

```diff
@@ -52,17 +52,27 @@ class FiscalInformationController @Inject()(
         .flatMap(_.procedureCode)
         .exists(code => !ProcedureCodesData.osrProcedureCodes.contains(code))
 
-    if (fastForward && cacheContainsFiscalReferenceData) {
-      navigator.continueTo(mode, routes.AdditionalFiscalReferencesController.displayPage(_, itemId))
-    } else if (fastForward && cacheItemIneligibleForOSR) {
-      navigator.continueTo(mode, routes.ProcedureCodesController.displayPage(_, itemId))
-    } else {
+    def displayFiscalInformationPage() = {
       val frm = form().withSubmissionErrors()
       request.cacheModel.itemBy(itemId).flatMap(_.fiscalInformation) match {
         case Some(fiscalInformation) => Ok(fiscalInformationPage(mode, itemId, frm.fill(fiscalInformation)))
         case _                       => Ok(fiscalInformationPage(mode, itemId, frm))
       }
     }
+
+    if (mode == Mode.Change) {
+      displayFiscalInformationPage()
+    } else if (fastForward && cacheContainsFiscalReferenceData) {
+      navigator.continueTo(mode, routes.AdditionalFiscalReferencesController.displayPage(_, itemId))
+    } else if (fastForward && cacheItemIneligibleForOSR) {
+      navigator.continueTo(mode, routes.ProcedureCodesController.displayPage(_, itemId))
+    } else {
+      if (cacheContainsFiscalReferenceData) {
+        navigator.continueTo(mode, controllers.declaration.routes.AdditionalFiscalReferencesController.displayPage(_, itemId))
```

Comment:
@PatrykRudnicki @SteveSugden Do you happen to know what the `fastForward` flag is for? Also, I am right to say that this case is redundant? 
It is executed if `cacheContainsFiscalReferenceData` is `true`, no matter what is the value for `fastForward`.

Reply:
I don't remember but it was something with summary pages probably.

I think it's necessary.

The easiest way to find out what this is doing is to search in the codebase where we're setting fastForward to true

Reply:
I think it was something to do with making the back-button work correctly.  But I can't remember the details.  But if you break it, I think there are ATs for it.

Reply:
It is only used once in `Navigator `(set to true):
`case CommodityDetails     => controllers.declaration.routes.FiscalInformationController.displayPage(_, _, fastForward = true)`

Reply:
Although I am still not sure about the intent. Thank you for AT hint. may try it later.